src/main/java/randoop/generation/InputSequenceSelector.java

```diff
@@ -0,0 +1,24 @@
+package randoop.generation;
+
+import randoop.sequence.ExecutableSequence;
+import randoop.sequence.Sequence;
+import randoop.util.SimpleList;
+
+/** Interface for selecting sequences as input for creating new sequences. */
+public interface InputSequenceSelector {
+  /**
+   * Choose a sequence used as input for creating a new sequence.
+   *
+   * @param candidates sequences to choose from
+   * @return the chosen sequence
+   */
+  Sequence selectInputSequence(SimpleList<Sequence> candidates);
+
+  /**
+   * Make use of the given {@link ExecutableSequence} to eventually update the weight map using
```

Comment:
This comment is about updating the weight map, but the name is about execution time.  I suggest making them consistent, or improving the documentation.